proselint/checks/misc/but.py

```diff
@@ -14,13 +14,31 @@
 
 """
 
-from proselint.registry.checks import Check, types
+import re
+from collections.abc import Iterator
+
+from proselint.registry.checks import Check, CheckResult
+
+
+def _check_but_paragraphs(text: str, check: Check) -> Iterator[CheckResult]:
+    pattern = r"(^|^\n|\n\n)But\b"
+    for m in re.finditer(pattern, text):
+        count = m.group(0).count("\n", 0, 2)
+        yield CheckResult(
+            start_pos=m.start() + count + check.offset[0],
+            end_pos=check.offset[1],
+            check_path=check.path,
+            message=check.message,
+            replacements=None,
+        )
```

Comment:
This is equivalent to running an `ExistenceSimple` check. It would be preferable to return an iterator expression regardless, due to performance improvements for Python 3.13+.